test/Microsoft.ML.Functional.Tests/ModelLoading.cs

```diff
@@ -74,35 +81,49 @@ public void LoadModelAndExtractPredictor()
             {
                 // This model can be loaded either as a composite data loader,
                 // a transformer model + an input schema, or a transformer model + a data loader.
-                var t = _ml.Model.LoadWithDataLoader(fs, out IDataLoader<IMultiStreamSource> l);
+                var t = _ml.Model.LoadWithDataLoader(fs, out loadedCompositeLoader);
+                // This is a bit strange, as it seems to test that it can reload from the same
+                // stream twice opened only once, which as far as I know is not really a requirement
+                // of the design or API, but we are nonetheless testing it. If this winds up failing,
+                // I'm not sure we should really insist on this as a design requirement.
                 var t1 = _ml.Model.Load(fs, out var s);
-                loadedCompositeLoader = _ml.Model.Load(fs);
+
+                CheckSameSchemas(loadedCompositeLoader.GetOutputSchema(), s);
+                // We combined the GAM with the loader, so the remaining chain should just be empty.
+                Assert.Empty(Assert.IsType<TransformerChain<ITransformer>>(t));
+                Assert.Empty(Assert.IsType<TransformerChain<ITransformer>>(t1));
             }
             using (var fs = File.OpenRead(loaderAndTransformerModelPath))
             {
                 // This model can be loaded either as a composite data loader,
                 // a transformer model + an input schema, or a transformer model + a data loader.
                 var t = _ml.Model.Load(fs, out var s);
-                var c = _ml.Model.Load(fs);
-                loadedTransformerModel1 = _ml.Model.LoadWithDataLoader(fs, out IDataLoader<IMultiStreamSource> l);
+                CheckSameSchemas(loader.GetOutputSchema(), s);
+
+                loadedTransformerModel1 = _ml.Model.LoadWithDataLoader(fs, out var l);
             }
 
-            var gam = ((loadedTransformerModel as ISingleFeaturePredictionTransformer<object>).Model
-                as CalibratedModelParametersBase).SubModel
-                as GamBinaryModelParameters;
-            Assert.NotNull(gam);
-
-            gam = (((loadedCompositeLoader as CompositeDataLoader<IMultiStreamSource, ITransformer>).Transformer.LastTransformer
-                as ISingleFeaturePredictionTransformer<object>).Model
-                as CalibratedModelParametersBase).SubModel
-                as GamBinaryModelParameters;
-            Assert.NotNull(gam);
-
-            gam = (((loadedTransformerModel1 as TransformerChain<ITransformer>).LastTransformer
-                as ISingleFeaturePredictionTransformer<object>).Model
-                as CalibratedModelParametersBase).SubModel
-                as GamBinaryModelParameters;
-            Assert.NotNull(gam);
+            void AssertIsGam(ITransformer trans)
```

Comment:
>void [](start = 12, length = 4)

Interesting. I didn't know this can be done :-).

Reply:
New in one of the c# 7 versions... Good for tests especially I've found when you have lots of this only locally useful but often repeated here code.